Build failure: E0599 – `.as_ref()` not found for PackageNameMatcher in `resolvo/mod.rs`

### Checklist

- [x] I added a descriptive title
- [x] I searched open reports and couldn't find a duplicate

### What happened?

The project fails to compile due to the following error:
```bash
error[E0599]: no method named `as_ref` found for enum `PackageNameMatcher` in the current scope
  --> crates/rattler_solve/src/resolvo/mod.rs:560
```
The code attempts to call:
```rust
spec.name.as_ref()?;
```
However, PackageNameMatcher does not implement .as_ref(), which causes a compilation failure.

This results in a hard build break and prevents the project from compiling successfully.

### What should have happened instead?

The project should compile successfully without method resolution errors.

Based on the enum implementation, .as_ref() appears to be invalid for PackageNameMatcher. The appropriate method seems to be:`.as_exact()`
Replacing .as_ref() with .as_exact() resolves the compilation error.


### Additional Context

File: `crates/rattler_solve/src/resolvo/mod.rs`
- Line: 560
- Error code: E0599
- Severity: High (breaks build)

#### Suggested fix:
Replace:
```rust
.as_ref()
```
With:
```rust
.as_exact()
```
This restores successful compilation.

--- a/crates/rattler_solve/src/resolvo/mod.rs
+++ b/crates/rattler_solve/src/resolvo/mod.rs
@@ -557,10 +557,8 @@ impl<'a> CondaDependencyProvider<'a> {
         // Build a lookup table for dependency overrides keyed by target package name.
         let mut override_map: HashMap<PackageName, Vec<DependencyOverride>> = HashMap::new();
         for rule in dependency_overrides {
-            if let Some(name) = rule.override_spec.name.as_ref() {
-                if let Some(name) = Option::<PackageName>::from(name.clone()) {
-                    override_map.entry(name).or_default().push(rule);
-                }
+            if let Some(name) = rule.override_spec.name.as_exact() {
+                override_map.entry(name.clone()).or_default().push(rule);
             }
         }
 
@@ -600,9 +598,8 @@ impl<'a> CondaDependencyProvider<'a> {
         record: &RepoDataRecord,
         dep_spec: &MatchSpec,
     ) -> Option<String> {
-        let dep_name = dep_spec.name.as_ref()?;
-        let dep_name = Option::<PackageName>::from(dep_name.clone())?;
-        let rules = self.dependency_overrides.get(&dep_name)?;
+        let dep_name = dep_spec.name.as_exact()?;
+        let rules = self.dependency_overrides.get(dep_name)?;
         for rule in rules {
             if rule.package_matcher.matches(&record.package_record) {
                 return Some(rule.override_spec.to_string());

--- a/py-rattler/src/index_json.rs
+++ b/py-rattler/src/index_json.rs
@@ -227,7 +227,7 @@ impl PyIndexJson {
 
     /// Track features are nowadays only used to down-weigh packages (ie. give them less priority). To
     /// that effect, the number of track features is counted (number of commas) and the package is downweighted
-    /// by the number of track_features.
+    /// by the number of `track_features`.
     #[getter]
     pub fn track_features(&self) -> Vec<String> {
         self.inner.track_features.clone()

--- a/py-rattler/src/nameless_match_spec.rs
+++ b/py-rattler/src/nameless_match_spec.rs
@@ -132,12 +132,12 @@ impl PyNamelessMatchSpec {
         self.inner.sha256.map(|sha256| PyBytes::new(py, &sha256))
     }
 
-    /// Returns a string representation of MatchSpec
+    /// Returns a string representation of `MatchSpec`
     pub fn as_str(&self) -> String {
         format!("{}", self.inner)
     }
 
-    /// Match a PyNamelessMatchSpec against a PackageRecord
+    /// Match a `PyNamelessMatchSpec` against a `PackageRecord`
     pub fn matches(&self, record: &PyRecord) -> bool {
         self.inner.matches(&record.as_package_record().clone())
     }

--- a/py-rattler/src/networking/cached_repo_data.rs
+++ b/py-rattler/src/networking/cached_repo_data.rs
@@ -26,7 +26,7 @@ impl From<CachedRepoData> for PyCachedRepoData {
 
 #[pymethods]
 impl PyCachedRepoData {
-    /// Returns a string representation of PyCachedRepoData.
+    /// Returns a string representation of `PyCachedRepoData`.
     pub fn as_str(&self) -> String {
         format!("{:?}", self.inner)
     }

--- a/py-rattler/src/paths_json.rs
+++ b/py-rattler/src/paths_json.rs
@@ -367,7 +367,7 @@ impl PyPrefixPlaceholder {
     }
 
     /// The type of the file, either binary or text. Depending on the type of file either text
-    /// replacement is performed or CString replacement.
+    /// replacement is performed or `CString` replacement.
     #[getter]
     pub fn file_mode(&self) -> PyFileMode {
         self.inner.file_mode.into()

--- a/py-rattler/src/prefix_paths.rs
+++ b/py-rattler/src/prefix_paths.rs
@@ -69,8 +69,8 @@ impl From<PrefixPaths> for PyPrefixPaths {
     }
 }
 
-/// An entry in the paths_data attribute of the PrefixRecord
-/// This is similar to PathsEntry from paths_json but refers
+/// An entry in the `paths_data` attribute of the `PrefixRecord`
+/// This is similar to `PathsEntry` from `paths_json` but refers
 /// to an entry for an installed package
 #[pyclass]
 #[repr(transparent)]
@@ -92,7 +92,7 @@ impl From<PyPrefixPathsEntry> for PathsEntry {
 }
 
 /// The path type of the path entry
-/// This is similar to PathType from paths_json; however, it contains additional enum fields
+/// This is similar to `PathType` from `paths_json`; however, it contains additional enum fields
 /// since it represents a file that's installed
 #[pyclass]
 #[repr(transparent)]
@@ -249,7 +249,7 @@ impl PyPrefixPathsEntry {
     }
 
     /// A hex representation of the SHA256 hash of the contents of the file
-    /// If prefix_placeholder is present, this represents the hash of the file *before*
+    /// If `prefix_placeholder` is present, this represents the hash of the file *before*
     /// any placeholders were replaced
     #[getter]
     pub fn sha256<'a>(&self, py: Python<'a>) -> Option<Bound<'a, PyBytes>> {
@@ -262,7 +262,7 @@ impl PyPrefixPathsEntry {
     // }
 
     /// A hex representation of the SHA256 hash of the contents of the file as installed
-    /// This will be present only if prefix_placeholder is defined. In this case,
+    /// This will be present only if `prefix_placeholder` is defined. In this case,
     /// this is the hash of the file after the placeholder has been replaced.
     #[getter]
     pub fn sha256_in_prefix<'a>(&self, py: Python<'a>) -> Option<Bound<'a, PyBytes>> {

--- a/py-rattler/src/record.rs
+++ b/py-rattler/src/record.rs
@@ -35,7 +35,7 @@ use crate::{
 /// It uses a `RecordInner` enum and (try_)as_{x}_record methods for this
 /// interface.
 ///
-/// PyO3 cannot expose tagged enums directly, to achieve this we use the
+/// `PyO3` cannot expose tagged enums directly, to achieve this we use the
 /// `PyRecord` wrapper pyclass on top of `RecordInner`.
 #[pyclass]
 #[repr(transparent)]
@@ -531,7 +531,7 @@ impl PyRecord {
     /// Track features are nowadays only used to downweight packages
     /// (ie. give them less priority). To that effect, the number of track
     /// features is counted (number of commas) and the package is downweighted
-    /// by the number of track_features.
+    /// by the number of `track_features`.
     #[getter]
     pub fn track_features(&self) -> Vec<String> {
         self.as_package_record().track_features.clone()
@@ -668,7 +668,7 @@ impl PyRecord {
 
     /// The spec that was used when this package was installed. Note that this
     /// field is not updated if the currently another spec was used.
-    /// Deprecated: Use requested_specs instead.
+    /// Deprecated: Use `requested_specs` instead.
     #[getter]
     #[allow(deprecated)]
     pub fn requested_spec(&self) -> PyResult<Option<String>> {
@@ -683,7 +683,7 @@ impl PyRecord {
     }
 
     /// Multiple specs that were used when this package was installed.
-    /// This field replaces the deprecated requested_spec field.
+    /// This field replaces the deprecated `requested_spec` field.
     #[getter]
     pub fn requested_specs(&self) -> PyResult<Vec<String>> {
         Ok(self.try_as_prefix_record()?.requested_specs.clone())
@@ -810,7 +810,7 @@ impl AsMut<PackageRecord> for PyRecord {
 
 #[pymethods]
 impl PyRecord {
-    /// Parses a PrefixRecord from a file.
+    /// Parses a `PrefixRecord` from a file.
     #[staticmethod]
     pub fn from_path(path: PathBuf) -> PyResult<Self> {
         Ok(PrefixRecord::from_path(path)

--- a/py-rattler/src/repo_data/mod.rs
+++ b/py-rattler/src/repo_data/mod.rs
@@ -46,7 +46,7 @@ impl PyRepoData {
 
 #[pymethods]
 impl PyRepoData {
-    /// Parses RepoData from a file.
+    /// Parses `RepoData` from a file.
     #[staticmethod]
     pub fn from_path(path: PathBuf) -> PyResult<Self> {
         Ok(RepoData::from_path(path)

--- a/py-rattler/src/run_exports_json.rs
+++ b/py-rattler/src/run_exports_json.rs
@@ -129,9 +129,9 @@ impl PyRunExportsJson {
         self.inner.strong = strong;
     }
 
-    /// NoArch run exports apply a run export only to noarch packages (other run exports are ignored).
+    /// `NoArch` run exports apply a run export only to noarch packages (other run exports are ignored).
     /// For example, python uses this to apply a dependency on python to all noarch packages, but not to
-    /// the python_abi package.
+    /// the `python_abi` package.
     #[getter]
     pub fn noarch(&self) -> Vec<String> {
         self.inner.noarch.clone()

--- a/py-rattler/src/solver.rs
+++ b/py-rattler/src/solver.rs
@@ -60,9 +60,9 @@ impl PyMinimumAgeConfig {
     /// Args:
     ///     seconds: The minimum age in seconds that a package must have been
     ///         published before it can be installed.
-    ///     exempt_packages: Optional list of package names that are exempt
+    ///     `exempt_packages`: Optional list of package names that are exempt
     ///         from the minimum age requirement.
-    ///     include_unknown_timestamp: Whether to include packages without a
+    ///     `include_unknown_timestamp`: Whether to include packages without a
     ///         timestamp. Defaults to False (exclude them).
     #[new]
     #[pyo3(signature = (seconds, exempt_packages=None, include_unknown_timestamp=false))]

--- a/py-rattler/src/version/version_spec.rs
+++ b/py-rattler/src/version/version_spec.rs
@@ -27,7 +27,7 @@ impl From<PyVersionSpec> for VersionSpec {
 
 #[pymethods]
 impl PyVersionSpec {
-    /// Construct a new VersionSpec from a string with optional strict parsing.
+    /// Construct a new `VersionSpec` from a string with optional strict parsing.
     #[new]
     #[pyo3(signature = (spec, strict=false))]
     pub fn __init__(spec: &str, strict: bool) -> PyResult<Self> {

